crypto/src/main/scala/org/bitcoins/crypto/StringFactory.scala

```diff
@@ -0,0 +1,20 @@
+package org.bitcoins.crypto
+
+import scala.util.Try
+
+/** A common factory trait that can be re-used to deserialize a string to a type t */
+trait StringFactory[T] {
+
+  /** Tries to parse a string to type t, throws an exception if fails */
+  def fromString(string: String): T
+
+  /** Treis to parse a string to type t, returns None if failure */
+  def fromStringOpt(string: String): Option[T] = {
+    fromStringT(string).toOption
+  }
+
+  /** Tries to parse string to type t, returns [[scala.util.Failure]] if the fails */
+  def fromStringT(string: String): Try[T] = {
+    Try(fromString(string))
+  }
```

Comment:
This seems backwards to me, I think that users should implement this and that `fromString` should default to `fromStringT.getOrElse(throw something)`

Reply:
specifically it should probably throw the failure:
```
def fromString(string: String): T = {
  fromStringT(string) match {
    case Success(t) => t
    case Failure(err) => throw err
  }
}

Reply:
My thinking here is the default implementation -- that must be implemented if you want parsing functionality -- is the unchecked exception version. Since this is the lowest common denominator I figured this would be best left unimplemented. If you want to totally ignore the safe APIs you can. 

If you want to use safe APIs that return `Try` or `Option` you can still override the respective methods and implement `fromString` as you have said.